model/GamutCompress.nk

```diff
@@ -96,68 +83,31 @@ set N59f7d000 [stack 0]
-  name Switch_hexagonal
-  note_font Helvetica
-  xpos -40
-  ypos 15
+  ypos -82
  }
  Group {
-  name compress
-  knobChanged "from math import (pow, exp, log, tan, atan, atanh, cos, pi, fabs)\n\ndef i_log(x):\n    return (exp((1-t+t*log(1-x)-x*t*log(1-x))/(t*(1-x))))*t+x*t-k\ndef i_exp(x):\n    return -log((-x+1)/(t-x))*(-t+x)+t-k\ndef i_atan(x):\n    return (2*tan((pi*(1-t))/(2*(x-t)))*(x-t))/pi+t-k\ndef i_tanh(x):\n    return atanh((1-t)/(x-t))*(x-t)+t-k\ndef sign(x):\n    return 0 if x == 0 else 1 if x > 0 else -1\n\ndef bisect(f, a, b):\n    if sign(f(a)) == sign(f(b)): \n        # bad estimate. return something close to linear\n        if method == 'exp' or method == 'log':\n            return -100\n        else: \n            return 1.99999\n    c = (a+b)/2.0\n    y = f(c)\n    if fabs(y) <= tol:\n        return c # lucky guess\n    n = 1\n    while fabs(y) > tol and n <= nmax:\n        if sign(y) == sign(f(a)):\n            a = c\n        else:\n            b = c\n        c = (a+b)/2.0\n        y = f(c)\n        n += 1\n    return c\n\ndef calc():\n    a = 1.0001\n    b = 5\n    if method == 'log':\n        f = i_log\n        a = -50\n        b = 0.98\n    elif method == 'exp':\n        f = i_exp\n    elif method == 'atan':\n        f = i_atan\n    elif method == 'tanh':\n        f = i_tanh\n        a = 1.000001\n    lim = bisect(f, a, b)\n    return lim\n\nn = nuke.thisNode()\nknob = nuke.thisKnob()\n\nn.begin()\nmeth = n\['method'].value()\nif meth != 'reinhard' and meth != 'power':\n    tol = 1e-3\n    nmax = 100\n    \n    tk = n\['threshold']\n    if tk.singleValue():\n        thr = \[tk.getValue()]*3\n    else:\n        thr = \[v for v in tk.getValue()]\n\n    method = n\['method'].value()\n\n    k_knobs = \['cyan', 'magenta', 'yellow']\n    lim_knobs = \['lim_x', 'lim_y', 'lim_z']\n    if knob.name() in k_knobs:\n        i = k_knobs.index(knob.name())\n        lim_knob = lim_knobs\[i]\n        t = thr\[i]\n        k = knob.getValue()+1\n        lim = calc()\n        n\[lim_knob].setValue(lim)\n    elif knob.name() == 'method' or knob.name() == 'threshold':\n        for i, lknob in enumerate(lim_knobs):\n            k = n\[k_knobs\[i]].getValue()+1\n            t = thr\[i]\n            lim = calc()\n            n\[lknob].setValue(lim)\nn.end()"
-  xpos -40
-  ypos 62
+  name compress1
+  xpos 180
+  ypos 63
   addUserKnob {20 compress}
-  addUserKnob {4 method M {log reinhard power exp atan tanh "" ""}}
-  method power
   addUserKnob {18 threshold R 0.4 1}
-  threshold 0.8
+  threshold {{parent.compress.threshold}}
   addUserKnob {6 threshold_panelDropped l "panel dropped state" -STARTLINE +HIDDEN}
   addUserKnob {7 power R 1 3}
-  power 1.2
+  power {{parent.compress.power}}
   addUserKnob {7 cyan}
-  cyan 0.09
+  cyan {{parent.compress.cyan}}
   addUserKnob {7 magenta}
-  magenta 0.24
+  magenta {{parent.compress.magenta}}
   addUserKnob {7 yellow}
-  yellow 0.12
+  yellow {{parent.compress.yellow}}
   addUserKnob {26 ""}
```

Comment:
Now that we've removed the bisection solver, the `lim_x, lim_y, lim_z` knobs don't do anything anymore and could be removed.